presto-main/src/main/java/com/facebook/presto/cost/PlanNodeCostEstimate.java

```diff
@@ -108,20 +165,27 @@ public boolean equals(Object o)
             return false;
         }
         PlanNodeCostEstimate that = (PlanNodeCostEstimate) o;
-        return Double.compare(that.cpuCost, cpuCost) == 0 &&
+        return Double.compare(that.individualCpuCost, individualCpuCost) == 0 &&
+                Double.compare(that.individualMemoryCost, individualMemoryCost) == 0 &&
+                Double.compare(that.individualNetworkCost, individualNetworkCost) == 0 &&
+                Double.compare(that.cpuCost, cpuCost) == 0 &&
                 Double.compare(that.memoryCost, memoryCost) == 0 &&
                 Double.compare(that.networkCost, networkCost) == 0;
     }
 
     @Override
     public int hashCode()
     {
-        return Objects.hash(cpuCost, memoryCost, networkCost);
+        return Objects.hash(individualCpuCost, individualMemoryCost, individualNetworkCost,
+                cpuCost, memoryCost, networkCost);
     }
 
     public PlanNodeCostEstimate add(PlanNodeCostEstimate other)
```

Comment:
@rschlussel @Cameronreynoldson Before this change `add` method was simply adding two estimates. E.g. there was not implicit assumption that the estimates being added are for the `source` nodes. In fact, this doesn't always seem to be the case. 

```
com.facebook.presto.cost.CostCalculatorWithEstimatedExchanges.ExchangeCostEstimator#visitAggregation

        @Override
        public PlanNodeCostEstimate visitAggregation(AggregationNode node, Void context)
        {
            PlanNodeStatsEstimate sourceStats = getStats(node.getSource());
            List<Symbol> sourceSymbols = node.getSource().getOutputSymbols();

            PlanNodeCostEstimate remoteRepartitionCost = CostCalculatorUsingExchanges.calculateExchangeCost(
                    numberOfNodes,
                    sourceStats,
                    sourceSymbols,
                    REPARTITION,
                    REMOTE,
                    types);
            PlanNodeCostEstimate localRepartitionCost = CostCalculatorUsingExchanges.calculateExchangeCost(
                    numberOfNodes,
                    sourceStats,
                    sourceSymbols,
                    REPARTITION,
                    LOCAL,
                    types);

            // TODO consider cost of aggregation itself, not only exchanges, based on aggregation's properties
            return remoteRepartitionCost.add(localRepartitionCost);
        }
```

Reply:
Would it be worth our effort to change the uses of PlanNodeCostEstimate to follow our new pattern of initializing with the individual costs and using add() for cumulative costs?